support/helm/helm.go

```diff
@@ -0,0 +1,177 @@
+	// for running the helm command in question
+	Args []string
+	// Wait is used to indicate if the helm command should wait for the runtime
+	// to reach an acceptable state before returning the control back. The duration
+	// for which this call is blocked is defined by the value set to Timeout
+	Wait bool
+	// Timeout is used to indicate the time to wait for any individual Kubernetes ops
+	Timeout string
+}
+
+type Manager struct {
+	e *gexe.Echo
+}
+
+type Option func(*Opts)
+
+const (
+	missingHelm = "'helm' command is missing. Please ensure the tool exists before using the helm manager"
+)
+
+func WithName(name string) Option {
+	return func(opts *Opts) {
+		opts.Name = name
+	}
+}
+
+func WithNamespace(namespace string) Option {
+	return func(opts *Opts) {
+		opts.Namespace = namespace
+	}
+}
+
+func WithReleaseName(releaseName string) Option {
+	return func(opts *Opts) {
+		opts.ReleaseName = releaseName
+	}
+}
+
+func WithVersion(version string) Option {
+	return func(opts *Opts) {
+		opts.Version = version
+	}
+}
+
+func WithChart(chart string) Option {
+	return func(opts *Opts) {
+		opts.Chart = chart
+	}
+}
+
+func WithMode(mode string) Option {
+	return func(opts *Opts) {
+		opts.Mode = mode
+	}
+}
+
+func WithArgs(args ...string) Option {
+	return func(opts *Opts) {
+		opts.Args = append(opts.Args, args...)
+	}
+}
+
+func WithWait() Option {
+	return func(opts *Opts) {
+		opts.Wait = true
+	}
+}
+
+func WithTimeout(timeout string) Option {
+	return func(opts *Opts) {
+		opts.Timeout = timeout
+	}
+}
+
+func (m *Manager) processOpts(opts ...Option) *Opts {
+	option := &Opts{}
+	for _, op := range opts {
+		op(option)
+	}
+	return option
+}
+
+func (m *Manager) getCommand(opt *Opts) (string, error) {
+	commandParts := []string{"helm", opt.Mode}
+	if opt.Mode == "" {
+		return "", fmt.Errorf("missing helm operation mode. Please use the WithMode option while invoking the Run")
+	}
+	if opt.Name != "" {
+		commandParts = append(commandParts, opt.Name)
+	}
+	if opt.Chart != "" {
+		commandParts = append(commandParts, opt.Chart)
+	} else {
+		commandParts = append(commandParts, opt.ReleaseName)
+	}
+	if opt.Namespace != "" {
+		commandParts = append(commandParts, "--namespace", opt.Namespace)
+	}
+	if opt.Version != "" {
+		commandParts = append(commandParts, "--version", opt.Version)
+	}
+	commandParts = append(commandParts, opt.Args...)
+	if opt.Wait {
+		commandParts = append(commandParts, "--wait")
+	}
+	if opt.Timeout != "" {
+		commandParts = append(commandParts, "--timeout", opt.Timeout)
+	}
+	return strings.Join(commandParts, " "), nil
+}
+
+func (m *Manager) Run(opts ...Option) (proc *exec.Proc, err error) {
+	if m.e.Prog().Avail("helm") == "" {
+		err = fmt.Errorf(missingHelm)
+		return
+	}
+	command, err := m.getCommand(m.processOpts(opts...))
+	if err != nil {
+		return
+	}
+	log.V(4).InfoS("Running Helm Operation", "command", command)
+	proc = m.e.RunProc(command)
+	return
+}
+
+func NewManager() *Manager {
```

Comment:
When you go to use this in an example I suspect you will want to  a way to pass in a kubeconfig reference, so that helm uses the right config generated by the test suite without an optional flag.

I am not sure if that should be required by default (e.g. in this initializer) or as an additional option.

Reply:
Yes. You are ready gut about that.. I am amusing then`WithArgs` to achieve that for now.. But I can extend the new object creation so that kube config is embedded into the helm object and avoid end user passing that one. Nice catch